### PR TITLE
🔁 Swaps UI

### PR DESCRIPTION
**Pull request by shivekkhurana**
### Summary

Introduce all UI components for swaps.

This is a part of the larger feature for integration of Paraswap in the wallet section. 
Related to: https://github.com/status-im/status-react/issues/12523

<details>
  <summary>📸 Screenshots</summary>

Simulator Screen Shot - iPhone 11 - 2022-01-05 at 18 02 23

Simulator Screen Shot - iPhone 11 - 2022-01-05 at 18 02 26

Simulator Screen Shot - iPhone 11 - 2022-01-05 at 18 02 35

Simulator Screen Shot - iPhone 11 - 2022-01-05 at 18 02 39

Simulator Screen Shot - iPhone 11 - 2022-01-05 at 18 02 43

Simulator Screen Shot - iPhone 11 - 2022-01-05 at 18 02 50

</details>

### Review notes
This is just the UI and some re-frame effects. The contract authentication is not in place yet.

### Testing notes
Not to be tested at the moment. It's hidden behind a feature flag, only visible in dev mode.

#### Platforms
- Android
- iOS

#### Areas that maybe impacted
- wallet

##### Functional
- wallet / transactions

### Steps to test
- Not to test at the moment. Is hidden behind a ff.

status: ready

### CODE REVIEW

File: src/quo/components/list/item.cljs [hunk 1]
```diff
@@ -266,3 +268,4 @@
         [text/text {:color :negative
                     :size  :small}
          error]])]))
+;; => #'quo.components.list.item/list-item
```
Comment (flexsurfer) [inline src/quo/components/list/item.cljs]: ?

File: src/quo/components/list/item.cljs [hunk 2]
```diff
@@ -223,7 +225,7 @@
      [component
       (merge {:type                :list-item
               :disabled            disabled
-              :bg-color            active-background
+              :bg-color            (when active-background-enabled active-background)
```
Comment (flexsurfer) [inline src/quo/components/list/item.cljs:228]: why this is needed?

File: src/status_im/ui/screens/wallet/account/views.cljs [hunk 3]
```diff
@@ -102,20 +103,19 @@
...
-                  mainnet? [:mainnet?]
                   ethereum-network? [:ethereum-network?]]
     (let [{:keys [tab]} @state]
       [react/view {:flex 1}
        [react/view {:flex-direction :row :margin-bottom 8 :padding-horizontal 4}
         [tabs/tab-title state :assets (i18n/label :t/wallet-assets) (= tab :assets)]
         (when ethereum-network?
-          [tabs/tab-title state :nft (i18n/label :t/wallet-collectibles) (= tab :nft)])
-        [tabs/tab-title state :history (i18n/label :t/history) (= tab :history)]]
+          [:<>
```
Comment (flexsurfer) [inline src/status_im/ui/screens/wallet/account/views.cljs]: how is this change related to PR ?

**Comment by du82** [comment]
Why does it say "Powered by Paraswap" at the top? Seems unnecessary and doesn't really fit the self-sufficient theme of Status.

File: src/quo/components/list/item.cljs (hunk 2 above)
Comment (shivekkhurana) [reply src/quo/components/list/item.cljs:228]: Carl has made updates to the design system. We now have cards ui. I tried to re-purpose the list-item to act as a card. The only problem was that `card`s have rounded corners but `list-item`s don't.

This change removes the on-press background color. If this doesn't exist, this is what the on-press interaction will look like (observe the odd borders, because it doesn't fill the container).

https://user-images.githubusercontent.com/1925158/148245458-c0878c7f-0123-4035-9148-fb63b36c789d.mov

File: src/status_im/ui/screens/wallet/account/views.cljs (hunk 3 above)
Comment (shivekkhurana) [reply src/status_im/ui/screens/wallet/account/views.cljs]: This must have slipped in while I was resolving conflicts. This is not related to the PR.

**Comment by shivekkhurana** [comment]
@ducheng0 

We need to ensure that users realise that they are using a third-party service, for legal reasons. 

There will also be an opt-in similar to the opensea and link unfurling opt-ins, so the user knows they are interacting with a 3rd party service (and that they might leak some metadata).

**Review by flexsurfer — APPROVED** [review]

File: src/status_im/ui/screens/wallet/account/views.cljs [hunk 4]
```diff
@@ -181,10 +180,33 @@
...
+(defn top-actions []
+  [react/view {:style styles/top-actions}
+   [round-action-button {:icon     :main-icons/add
+                         :title    (i18n/label :t/buy-crypto)
+                         :on-press #(re-frame/dispatch [:buy-crypto.ui/open-screen])}]
+   [round-action-button {:icon  :main-icons/change
+                         :title (i18n/label :t/swap)
+                         :on-press #(re-frame/dispatch [:open-modal :token-swap])}]])
+
+(defn floating-action [])
```
Comment (briansztamfater) [inline src/status_im/ui/screens/wallet/account/views.cljs]: Is this function used / needed?

Comment (shivekkhurana) [reply src/status_im/ui/screens/wallet/account/views.cljs]: No, it's a leftover. Removing it.
